sycl/source/detail/scheduler/commands.cpp

```diff
@@ -161,6 +226,190 @@ Command::Command(CommandType Type, QueueImplPtr Queue)
+        // Source node represents the event and this event needs
+        // to be completed before target node can execute
+        xpti_td *TgtEvent = (xpti_td *)MTraceEvent;
+        EdgeEvent->source_id = NodeEvent->unique_id;
+        EdgeEvent->target_id = TgtEvent->unique_id;
+        xptiAddMetadata(EdgeEvent, "event", EdgeName.c_str());
+        xptiNotifySubscribers(MStreamID, (uint16_t)xpti_tp::edge_create,
+                              cl::sycl::detail::g_sycl_graph_event, EdgeEvent,
+                              nullptr);
+      }
+    }
+  }
+#endif
+}
+
+void Command::makeTraceEventProlog(void *MAddress) {
+#ifdef XPTI_ENABLE_INSTRUMENTATION
+  MTraceEventPrologComplete = true;
+  // Setup the member variables with information needed for event notification
+  MCommandNodeType = commandToNodeType(MType);
+  MCommandName = commandToName(MType);
+  xpti::utils::StringHelper SH;
+  MAddressString = SH.addressAsString(MAddress);
+  std::string CommandString =
+      SH.nameWithAddressString(MCommandName, MAddressString);
+
+  xpti::payload_t p(CommandString.c_str(), MAddress);
+  xpti_td *CmdTraceEvent = xptiMakeEvent(
+      CommandString.c_str(), &p, (uint16_t)xpti_te::graph, xpti_at::active);
+  if (CmdTraceEvent) {
+    MTraceEvent = (void *)CmdTraceEvent;
+    // If we are seeing this event again, then the instance ID will be greater
+    // than 1; in this case, we must skip sending a notification to create a
+    // node as this node has already been created. We return this value so the
+    // epilog method can be called selectively.
+    MFirstInstance = (CmdTraceEvent->instance_id == 1);
+  }
+#endif
+}
+
+void Command::makeTraceEventEpilog() {
+#ifdef XPTI_ENABLE_INSTRUMENTATION
+  if (MTraceEventPrologComplete) {
+    if (MTraceEvent) {
+      xptiNotifySubscribers(
+          MStreamID, (uint16_t)xpti::trace_point_type_t::node_create,
+          cl::sycl::detail::g_sycl_graph_event, (xpti_td *)MTraceEvent,
+          static_cast<const void *>(MCommandNodeType.c_str()));
+    }
+  }
+#endif
+}
+
+void Command::addDep(DepDesc NewDep) {
+  if (NewDep.MDepCommand)
+    MDepsEvents.push_back(NewDep.MDepCommand->getEvent());
+  MDeps.push_back(NewDep);
+#ifdef XPTI_ENABLE_INSTRUMENTATION
+  emitEdgeEventForCommandDependence(
+      NewDep.MDepCommand, (void *)NewDep.MDepRequirement->MSYCLMemObj,
+      accessModeToString(NewDep.MDepRequirement->MAccessMode), true);
+#endif
+}
+
+void Command::addDep(EventImplPtr Event) {
+#ifdef XPTI_ENABLE_INSTRUMENTATION
+  // We need this for just the instrumentation, so
+  // guarding it will prevent unused variable warnings
+  // when instrumentation is turned off
+  Command *Cmd = (Command *)Event->getCommand();
+  RT::PiEvent &PiEventAddr = Event->getHandleRef();
+#endif
+
+  MDepsEvents.push_back(std::move(Event));
+
+#ifdef XPTI_ENABLE_INSTRUMENTATION
+  // Now make an edge for the dependent event
+  emitEdgeEventForEventDependence(Cmd, PiEventAddr);
```

Comment:
Can't you move the call to the first ifdef section in the function?

Reply:
Yes, and it has been done.